lib/std/Thread.zig

```diff
@@ -517,8 +517,8 @@ pub fn cpuCount() CpuCountError!usize {
         },
         .haiku => {
             var count: u32 = undefined;
-            var system_info: os.system_info = undefined;
-            const rc = os.system.get_system_info(&system_info);
+            // var system_info: os.system_info = undefined;
+            // const rc = os.system.get_system_info(&system_info);
```

Comment:
@hoanga Please introduce the `B_*` constants for error codes. Returning 1 here is a safe default but I think errors should be bubbled up to the caller.
Speaking of `get_system_info`, this [line](https://github.com/haiku/haiku/blob/e41899a6fefb89519b17c980bf5230d8e66e3fe6/src/system/libroot/os/system_info.cpp#L97) looks wrong, they probably meant `return error`.

```zig
const count: u32 = if (os.system.get_system_info(&system_info) != 0)
    system_info.cpu_count
else
    1;
```